R/caretStack.R

```diff
@@ -229,6 +250,41 @@ check_caretStack <- function(object) {
   )
 }
 
+#' @title Check original_features parameter
+#' @description Check the original_features parameter for caretStack, predict.caretStack and varImp.caretStack
+#'
+#' @param original_features a character vector with the names of the original features to include in the stack
+#' @param newdata the data to use for train a model or make predictions
+#' @param training a logical indicating whether the function is being called during training or prediction
+#' @keywords internal
+check_original_features <- function(original_features, newdata, training = TRUE) {
+  # Make sure newdata is not NULL when original_features is not NULL
+  if (!is.null(original_features) && is.null(newdata)) {
+    argument_name <- if (training) "new_X" else "newdata"
+    stop("original_features was specified, but ", argument_name, " is not provided. ",
+      "Please provide ", argument_name, " to use original features.",
+      call. = FALSE
+    )
+  }
+
+  if (is.character(original_features)) {
+    # Check that all specified variables in original_features are present in newdata
+    if (!all(original_features %in% colnames(newdata))) {
+      missing_columns <- setdiff(original_features, colnames(newdata))
+      argument_name <- if (training) "new_X" else "newdata"
```

Comment:
You could call this line once, before the if statements, instead of having it inside each if statement